### PR TITLE
[Cloud Client Library] Use V1 SDK in client

### PR DESCRIPTION
**Pull request by yawangwang**
Update the old version of client with the V1 cloud client library.

Use `go get cloud.google.com/go/confidentialcomputing/apiv1` to install the package.

### CODE REVIEW

File: launcher/verifier/rest/rest.go [hunk 1]
```diff
@@ -71,42 +83,42 @@ func NewClient(ctx context.Context, projectID string, region string, opts ...opt
 }
 
 type restClient struct {
-	service  *v1alpha1.Service
-	location *v1alpha1.Location
+	service  *v1.Client
```
Comment (alexmwu) [inline launcher/verifier/rest/rest.go:86]: nit: rename to something like v1Client

**Review by alexmwu — APPROVED** [review]
@jkl73 @yawangwang do we want experiment gating?

**Comment by yawangwang** [comment]
> @jkl73 @yawangwang do we want experiment gating?

Yes, I think experiment gating will give us more confidence in V1 sdk when rolling out changes.

**Review by jkl73 — COMMENTED** [review]
lgtm, we can talk about put this behind an experiment next.

File: launcher/verifier/rest/rest.go [hunk 2]
```diff
@@ -116,52 +128,48 @@ func convertChallengeFromREST(chal *v1alpha1.Challenge) (*verifier.Challenge, er
...
-			pcrVals[strIdx] = encoding.EncodeToString(val)
+			pcrVals[int32(idx)] = val
 		}
 
-		quotes[i] = &v1alpha1.Quote{
-			RawQuote:     encoding.EncodeToString(quote.GetQuote()),
-			RawSignature: encoding.EncodeToString(quote.GetRawSig()),
-			HashAlgo:     int64(quote.GetPcrs().GetHash()),
+		quotes[i] = &confidentialcomputingpb.TpmAttestation_Quote{
+			RawQuote:     quote.GetQuote(),
```
[the remark this answers is not in the document]
Comment (alexmwu) [reply launcher/verifier/rest/rest.go:145]: Yes, following https://google.aip.dev/140#string-vs-bytes since the bytes are already base 64 encoded

**Comment by alexmwu** [comment]
> lgtm, we can talk about put this behind an experiment next.

If we want it behind an experiment, we should probably make this change additive instead of also deleting the old client. Perhaps introduce a new client in restv1.go and remove the old one in a separate PR once we have completed experiment rollout.

**Comment by yawangwang** [comment]
> > lgtm, we can talk about put this behind an experiment next.
> 
> If we want it behind an experiment, we should probably make this change additive instead of also deleting the old client. Perhaps introduce a new client in restv1.go and remove the old one in a separate PR once we have completed experiment rollout.

Yes, there will be additional changes to this PR to ensure this client SDK change is behind an experiment. But for now this PR is blocked because it is dependent on Josh's client experiment changes.

**Review by alexmwu — APPROVED** [review]

**Pull request merged**